mysql-test/main/mysqladmin.test

```diff
@@ -51,3 +51,66 @@ EOF
 #
 --error 1
 --exec $MYSQLADMIN -u root -p 2>&1 > /dev/null
+
+#
+# MDEV-19168 Reload SSL certificate
+# From mysql-test/main/flush_ssl.test
+# This test reloads server SSL certs ./mysqladmin flush-ssl, and checks that 
+# 1. old SSL connections (that existed before FLUSH) still work and use old certificate
+# 2. new SSL connection use new certificate
+# 3. if ./mysqladmin flush-ssl runs into error, SSL is still functioning
+# SWtatus variable Ssl_server_not_after is used to tell the old certificate from new.
+#
+
+
+source include/have_ssl_communication.inc;
+
+# Restart server with cert. files located in temp directory
+# We are going to remove / replace them within the test, 
+# so we can't use the ones in std_data directly.
+
+let $ssl_cert=$MYSQLTEST_VARDIR/tmp/ssl_cert.pem;
+let $ssl_key=$MYSQLTEST_VARDIR/tmp/ssl_key.pem;
+
+copy_file $MYSQL_TEST_DIR/std_data/server-key.pem $ssl_key;
+copy_file $MYSQL_TEST_DIR/std_data/server-cert.pem $ssl_cert;
+
+let $restart_parameters=--ssl-key=$ssl_key --ssl-cert=$ssl_cert;
+--source include/kill_mysqld.inc
+--source include/start_mysqld.inc
+
+connect  ssl_con,localhost,root,,,,,SSL;
+SELECT VARIABLE_VALUE INTO @ssl_not_after FROM INFORMATION_SCHEMA.SESSION_STATUS WHERE VARIABLE_NAME='Ssl_server_not_after';
+let $ssl_not_after=`SELECT @ssl_not_after`;
+
+remove_file $ssl_cert;
+remove_file $ssl_key;
+
+--echo # Use a different certificate ("Not after" certificate field changed)
+copy_file $MYSQL_TEST_DIR/std_data/server-new-key.pem $ssl_key;
+copy_file $MYSQL_TEST_DIR/std_data/server-new-cert.pem $ssl_cert;
+
+--exec $MYSQLADMIN --default-character-set=latin1 -S $MASTER_MYSOCK -P $MASTER_MYPORT  -u root --password= flush-ssl 2>&1
+
+--echo # Check new certificate used by new connection
+exec $MYSQL --ssl  -e  "SELECT IF(VARIABLE_VALUE <> '$ssl_not_after', 'OK', 'FAIL') as Result FROM INFORMATION_SCHEMA.SESSION_STATUS WHERE VARIABLE_NAME='Ssl_server_not_after'";
```

Comment:
Given the `FLUSH SSL` feature is already thoroughly tested in `flush_ssl.test`, we don't need to duplicate all the code here. A simple test like this one where a new connection is tested to see if it uses the new certificate is enough to deduce that `mysqladmin` issues the right command to the server. In other words, we should stop at this line with the tests.

Reply:
the question is then why do we need to duplicate functionality of the 

> mysql -e "FLUSH SSL"

with new functionality in mysqladmin which exists mostly for legacy reasons, and for "ping" :)

Reply:
I don't have a good answer for this unfortunately, but because someone requested this option and someone implemented it, I don't see why I shouldn't give the contribution a chance.

Reply:
Heh, a different approach. I'd first ask if this makes a lot of sense, then give chance or not . Same thing as for developers.